lib/oli_web/live/community_live/index.ex

```diff
@@ -0,0 +1,120 @@
+defmodule OliWeb.CommunityLive.Index do
+  use Surface.LiveView, layout: {OliWeb.LayoutView, "live.html"}
+  use OliWeb.Common.SortableTable.TableHandlers
+
+  alias Oli.Accounts
+  alias Oli.Groups
+  alias OliWeb.Common.{Breadcrumb, Filter, Listing}
+  alias OliWeb.CommunityLive.{New, TableModel}
+  alias OliWeb.Router.Helpers, as: Routes
+  alias Surface.Components.Form
+  alias Surface.Components.Form.{Checkbox, Field, Label}
+  alias Surface.Components.Link
+
+  data title, :string, default: "Communities"
+  data breadcrumbs, :any
+
+  data filter, :any, default: %{"status" => "active"}
+  data query, :string, default: ""
+  data total_count, :integer, default: 0
+  data offset, :integer, default: 0
+  data limit, :integer, default: 20
+  data sort, :string, default: "sort"
+  data page_change, :string, default: "page_change"
+  data show_bottom_paging, :boolean, default: false
+  data additional_table_class, :string, default: ""
+
+  @table_filter_fn &__MODULE__.filter_rows/3
+  @table_push_patch_path &__MODULE__.live_path/2
+
+  def filter_rows(socket, query, filter) do
+    filter =
+      Enum.reduce(filter, %{}, fn {field, value}, acc ->
+        Map.put(acc, field, String.split(value, ","))
+      end)
+
+    query_str = String.downcase(query)
+    status_list = filter["status"]
+
+    Enum.filter(socket.assigns.communities, fn c ->
+      String.contains?(String.downcase(c.name), query_str) and
+        Enum.member?(status_list, Atom.to_string(c.status))
```

Comment:
Given that "status" is effectively user input (anyone can manually manipulate the query string), this implementation should probably filter the status list against a whitelist of what it knows to be valid statuses.  The potential for bad data (especially malicious) becomes more problematic when we consider that `Enum.member?` within `Enum.filter` is n^2.